src/libecalc/presentation/yaml/mappers/component_mapper.py

```diff
@@ -229,37 +233,41 @@ def from_yaml_to_dto(
-        self._target_period = target_period
-        self.__references = references
-
-    @staticmethod
-    def create_model(model: Dict, regularity: Dict[datetime, Expression]):
-        name = model.get(EcalcYamlKeywords.name, "")
-        user_defined_category = model.get(EcalcYamlKeywords.user_defined_tag, "")
-        emission_rate = model.get(EcalcYamlKeywords.installation_venting_emitter_emission_rate)
-
-        try:
-            return dto.EmitterModel(
-                name=name,
-                user_defined_category=user_defined_category,
-                emission_rate=emission_rate,
-                regularity=regularity,
-            )
-        except ValidationError as e:
-            raise DtoValidationError(data=model, validation_error=e) from e
-
-    def from_yaml_to_dto(
-        self, data: Optional[Dict], regularity: Dict[datetime, Expression]
-    ) -> Dict[datetime, dto.EmitterModel]:
-        time_adjusted_model = define_time_model_for_period(data, target_period=self._target_period)
-        return {
-            start_date: EmitterModelMapper.create_model(model, regularity=regularity)
-            for start_date, model in time_adjusted_model.items()
-        }
+# EmitterModelMapper should not be needed - commented out for check:
+
+# class EmitterModelMapper:
+#     """EmitterModel used by VentingEmitters."""
+#
+#     def __init__(self, references: References, target_period: Period):
+#         self._target_period = target_period
+#         self.__references = references
+#
+#     @staticmethod
+#     def create_model(model: Dict, regularity: Dict[datetime, Expression]):
+#         name = model.get(EcalcYamlKeywords.name, "")
+#         user_defined_category = model.get(EcalcYamlKeywords.user_defined_tag, "")
+#         emission_rate = model.get(EcalcYamlKeywords.installation_venting_emitter_emission_rate)
+#         emission_rate_type = model.get(EcalcYamlKeywords.venting_emitter_rate_type) or RateType.STREAM_DAY
+#         convert_to_stream_day = emission_rate_type == RateType.CALENDAR_DAY
+#         try:
+#             return dto.EmitterModel(
+#                 name=name,
+#                 user_defined_category=user_defined_category,
+#                 emission_rate=emission_rate,
+#                 regularity=regularity,
+#                 emission_rate_type=emission_rate_type,
+#             )
+#         except ValidationError as e:
+#             raise DtoValidationError(data=model, validation_error=e) from e
+#
+#     def from_yaml_to_dto(
+#         self, data: Optional[Dict], regularity: Dict[datetime, Expression]
+#     ) -> Dict[datetime, dto.EmitterModel]:
+#         time_adjusted_model = define_time_model_for_period(data, target_period=self._target_period)
+#         return {
+#             start_date: EmitterModelMapper.create_model(model, regularity=regularity)
+#             for start_date, model in time_adjusted_model.items()
+#         }
 
 
 class VentingEmittersMapper:
```

Comment:
Do we need a mapper?

Reply:
If you could have a look again, several changes. See if some of the `VentingEmittersMapper` should be moved? Now some of the "parsing" of the temporal emitter model is done here (earlier in the `YamlVentingEmitter` class)

Reply:
The mapper is removed